subsys/mgmt/hawkbit/Kconfig

```diff
@@ -79,6 +79,19 @@ config HAWKBIT_DDI_SECURITY_TOKEN
 	  Authentication security token for the configured hawkBit DDI
 	  authentication mode.
 
+config HAWKBIT_CUSTOM_DEVICE_ID
+	bool "Custom device id through callback function"
+	help
+	  Be able to customize the device id during runtime to a custom value,
+	  by configuring the callback function. See `hawkbit_set_device_identity_cb`
+
+config HAWKBIT_DEVICE_ID_MAX_LENGTH
+	int "Maximum length of the device id"
+	depends on HAWKBIT_CUSTOM_DEVICE_ID
+	default 32
+	help
```

Comment:
Could we have a range here, so that we can verify that the value is sane (I do not see it being verified in the code).